src/main.c

```diff
@@ -1031,6 +1044,14 @@ static gchar* info_formatter_json_base(RaucManifest *manifest, gboolean pretty)
 			json_builder_add_string_value(builder, "post-install");
 		}
 		json_builder_end_array(builder);
+		json_builder_set_member_name(builder, "incremental");
+		json_builder_begin_array(builder);
+		if (img->incremental) {
+			for (gchar *m = img->incremental[0]; m != NULL; m++) {
+				json_builder_add_string_value(builder, m);
```

Comment:
To not dump our memory, this should be

```suggestion
			for (gchar **m = img->incremental; *m != NULL; m++) {
				json_builder_add_string_value(builder, *m);
```

Also, if this key is compatibility-related, it is maybe worth adding a test?

Reply:
Fixed.